tests/unit/libbpf_test.cpp

```diff
@@ -2498,6 +2498,21 @@ TEST_CASE("BPF_PROG_BIND_MAP etc.", "[libbpf]")
     REQUIRE(map_fd2 > 0);
     Platform::_close(map_fd2);
 
+    //Verify we attach the program.
+    memset(&attr, 0, sizeof(attr));
+    attr.bpf_prog.prog_fd= program_fd;
+    attr.bpf_prog.target_fd = program_fd;
+    attr.bpf_prog.flags = 0;
+    attr.bpf_prog.prog_type = BPF_CGROUP_INET4_CONNECT;
```

Comment:
I see you changed this from XDP to this.  Does it matter?  If it's not supposed to work with XDP, is there a negative test somewhere?